File: src/main/java/com/busted_moments/client/models/territory/eco/types/EcoConstants.java
package com.busted_moments.client.models.territory.eco.types;

import com.busted_moments.client.models.territory.eco.TerritoryEco;

public interface EcoConstants {
   interface Tower {
      long DAMAGE_MAX = 1500;
      long DAMAGE_MIN = 1000;
      double ATTACK_SPEED = 0.5;
      long HEALTH = 300000;
      double DEFENSE = 10;
   }

   static long getResourceStorage(boolean isHQ) {
      return isHQ ? 1500 : 300;
   }

   static long getEmeraldStorage(boolean isHQ) {
      return isHQ ? 5000 : 3000;
   }

   static long getStorage(ResourceType type, TerritoryEco eco) {
      if (eco == null) return 0;
      var storage = (type == ResourceType.EMERALDS) ? getEmeraldStorage(eco.isHQ()) : getResourceStorage(eco.isHQ());

      return (long) (((eco.getUpgrade(type == ResourceType.EMERALDS ? UpgradeType.EMERALD_STORAGE : UpgradeType.RESOURCE_STORAGE).bonus() / 100) + 1) * storage);
   }
}

File: src/main/kotlin/com/busted_moments/client/models/territories/eco/EcoConstants.kt
package com.busted_moments.client.models.territories.eco

object EcoConstants {
    object Tower {
        const val DAMAGE_MAX: Long = 1500
        const val DAMAGE_MIN: Long = 1000
        const val ATTACK_SPEED: Double = 0.5
        const val HEALTH: Long = 300000
        const val DEFENSE: Double = 10.0
    }

    const val HQ_EMERALD_STORAGE = 5000
    const val NORMAL_EMERALD_STORAGE = 3000

    const val HQ_RESOURCE_STORAGE = 1500
    const val NORMAL_RESOURCE_STORAGE = 300

//    fun getStorage(type: ResourceType, eco: TerritoryEco?): Long {
//        if (eco == null) return 0
//        val storage =
//            if ((type == ResourceType.EMERALDS)) getEmeraldStorage(eco.isHQ()) else getResourceStorage(eco.isHQ())
//
//        return (((eco.getUpgrade(if (type == ResourceType.EMERALDS) UpgradeType.EMERALD_STORAGE else UpgradeType.RESOURCE_STORAGE)
//            .bonus() / 100) + 1) * storage)
//    }
}